### ISSUE TITLE
Do not ignore `type` `import/export` in favor of `eslint-import-resolver-typescript`

### ISSUE THREAD

**Issue by JounQin** [import-js/eslint-plugin-import#2500]
`type` `import/export` are always ignored now

```ts
import type { NonExisted } from 'package-a'
```

This is bad, `eslint-import-resolver-typescript` resolves `.d.ts`/`.ts` correctly, but they are ignored.

**Comment by JounQin** [comment]
Seems duplicate with #1675

**Comment by ljharb** [comment]
`tsc` should be catching this - does it not?

**Comment by JounQin** [comment]
> `tsc` should be catching this - does it not?

Right, a lot of rules for like `import/no-unresolved` can be caught be TypeScript itself, but tsc is heavier than a plain resolver.

**Comment by ljharb** [comment]
`tsc` runs at the same phase as a linter (design/build time, as well as in CI), and it doesn't make sense for a linter to be enforcing things that typescript does - at least when it adds complexity.

**Comment by JounQin** [comment]
> `tsc` runs at the same phase as a linter (design/build time, as well as in CI), and it doesn't make sense for a linter to be enforcing things that typescript does - at least when it adds complexity.

tsc and eslint are different things, and they can be used without each other. Resolving typings is expensive and slow, that’s why `esbuild`, `swc` ignore typings AFAIK.

**Comment by ljharb** [comment]
Sure - that both makes sense for a bundler, and is also why a linter would want to minimize needing to resolve typings.

**Comment by JounQin** [comment]
And are you meaning/saying we should not run this plugin at all on TypeScript files? It’s not right IMO.

This plugin has already been able to parse TypeScript files via `@typescript-eslint` and `type` nodes are already there, we just ignored them before. I imagine this would increase much size? Or maybe I can raise a PR for it?

**Comment by ljharb** [comment]
No, I'm not saying that at all. I'm saying we should not be duplicating checks that are already done in `tsc`. There's tons of things to lint that are specific to TS types but that aren't caught by TS because they're not type errors.

We explicitly and intentionally ignored `import type` statements awhile back, for this reason.

**Comment by JounQin** [comment]
I’m still confused, `import/no-unresolved` and many other rules can be covered by `tsc`, what’s the difference here then?

**Comment by ljharb** [comment]
tsc doesn't check things that aren't typed, especially within node_modules - and you could import a value from a file that isn't there that TS can't detect.

**Comment by JounQin** [comment]
> you could import a value from a file that isn't there that TS can't detect.

I don’t think so if you have correct typing declarations? And that’s why TypeScript exists more than just `Type`.

**Comment by ljharb** [comment]
Right - but many dependencies don't.

In general, if it's easy to catch something with eslint, even if TS catches it, then we should! However, if there's any complications with something that's duplicated, we should drop it - which is what we did with importing types (not just `import type`), because there were false positives due to the complexity of resolving typings.

**Comment by ljharb** [comment]
I'd have to dig them up. We still support the old version of the TS resolver, so I doubt it's been resolved there.
